### PR TITLE
DSP: Replace x64 JIT pointer with a pointer to an architecture-agnostic base

### PR DESCRIPTION
**Pull request by lioncash**
This adds a base class that is used to replace the concrete instance of the x64 JIT pointer within DSPCore. This fully removes the direct use (read: non-ifdefed) usage of x86-64-specifics within the main DSP code.

Said base can also be used for creating JITs for other architectures, such as AArch64, etc.

### CODE REVIEW

File: Source/Core/Core/DSP/Jit/DSPEmitterBase.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#endif
+
+namespace DSP::JIT
+{
+DSPEmitter::~DSPEmitter() = default;
+
+std::unique_ptr<DSPEmitter> CreateDSPEmitter()
+{
+#if defined(_M_X86) || defined(_M_X86_64)
+  return std::make_unique<x64::DSPEmitter>();
```
Comment (degasus) [inline Source/Core/Core/DSP/Jit/DSPEmitterBase.cpp]: Shouldn't here be the option if we want to use the DSP recompiler of the DSP interpreter?

Also the #else macro isn't required.

Comment (lioncash) [reply Source/Core/Core/DSP/Jit/DSPEmitterBase.cpp]: > Shouldn't here be the option if we want to use the DSP recompiler of the DSP interpreter?

I don't get what you mean by this

Comment (degasus) [reply Source/Core/Core/DSP/Jit/DSPEmitterBase.cpp]: I'm missing the
switch(opts.core_type)
{
#if X64
case DSPInitOptions::CoreType::JIT64: return std::make_unique<x64::DSPEmitter>();
#endif
case: interpreter: ...
default: std::make_unique<DSPEmitterNull>();

Comment (lioncash) [reply Source/Core/Core/DSP/Jit/DSPEmitterBase.cpp]: I don't get why the interpreter case needs to be handled here, it's not an emitter type (it's not even a class).

**Pull request merged**